### ISSUE TITLE
With Forces Matching If statement return types

### ISSUE THREAD

**Issue by daryllabar** [microsoft/Power-Fx#843]
This code works perfectly fine:
```
    If(IsBlank(note),
        Patch(SiteHistory, Defaults(SiteHistory), { TaskDetails: "Value" });
        Notify("New Note created", NotificationType.Information);
    ,
        Patch(SiteHistory, note, { TaskDetails: "Value" });
    );
```

But this code gives a very unhelpful error message of "The function If has some invalid arguments":
```
With({
    toUpdate: {
        TaskDetails: "Value"
    }
},
    If(IsBlank(note),
        Patch(SiteHistory, Defaults(SiteHistory), toUpdate);
        Notify("New Note created", NotificationType.Information);
    ,
        Patch(SiteHistory, note, toUpdate);
    );
);
```

But this works just fine:

```
With({
    toUpdate: {
        TaskDetails: "Value"
    }
},
    If(IsBlank(note),
        Notify("New Note created", NotificationType.Information);
        Patch(SiteHistory, Defaults(SiteHistory), toUpdate);
    ,
        Patch(SiteHistory, note, toUpdate);
    );
);
```

As well as this:
```
With({
    toUpdate: {
        TaskDetails: "Value"
    }
},
    If(IsBlank(note),
        Patch(SiteHistory, Defaults(SiteHistory), toUpdate);
        Notify("New Note created", NotificationType.Information);
    ,
        Patch(SiteHistory, note, toUpdate);
        Notify("Note updated", NotificationType.Information);
    );
);
```

Which leads me to believe that the If must return the same type of value within the With.  Is this expected?  If so, is this documented?  

I would expect that as long as I don't use the output of an if, it shouldn't care if different types of values are returned.

Session Details if helpful
> Session ID: e18c08cf-45bc-4662-87d1-2e5d548de70a
> PowerApps 3.22105.21
> Object ID: dc40ed33-a73f-4ffe-9c10-bf35b3cd21e8
> Tenant ID: d843cb99-8386-4d4c-a924-e7147b4e184a
> Embedded Correlation ID: 52450bb0-6463-11ed-a220-0fbb07ff730b
> App ID: c7eac419-9005-42bc-b262-ba75625182e6
> Telemetry location: global

**Comment by MikeStall** [comment]
"I would expect that as long as I don't use the output of an if, it shouldn't care if different types of values are returned." 
Agreed.  This might be related to https://github.com/microsoft/Power-Fx/blob/631173631d3b4ab9dd06d6fe2d3a2a2451c100cf/src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/If.cs#L169

**Comment by jas-valgotar** [comment]
This relates to fix #1113.
After the merge.
```
If(IsBlank(note),
        Patch(SiteHistory, Defaults(SiteHistory), { TaskDetails: "Value" });
        Notify("New Note created", NotificationType.Information);
    ,
        Patch(SiteHistory, note, { TaskDetails: "Value" });
    );
```
would show
`if(true, {test:1}, "mismatched args")` making it more clear to the user that the output of this can't really be used

**Comment by jas-valgotar** [comment]
This is Fixed via https://github.com/microsoft/Power-Fx/pull/1113
